### ISSUE TITLE
Contents of the sdk tarballs

### ISSUE THREAD

**Issue by hanetzer** [tpoechtrager/osxcross#4]
Many thanks for supplying the links, but, with my internet being as slow as it is, I'd rather not download something I more than likely already have; I have in my posession a tarball of the contents of the Xcode.app, obtained from my hackintosh install; would you be able to provide a list of the contents of the 10.9 sdk tarball (not en-totale, just a generic overview) to that effect?

**Comment by BioMike** [comment]
I have a related question regarding this. When going through the SDK tree there are in
target/SDK/MacOSX10.*.sdk/usr/{include, lib} a lot of libraries which could (should?) be build by the cross tool chain. Things like libz, libssl, etc. Would it be possible to setup the toolchain more like mxe (provide a package manager and have as much deps build by the cross tool chain as possible)? And use only the MacOSX specific libraries from the SDK?

I'm looking into cross building boost (I feel adventurous), but seeing the comments in MacPorts about broken stuff in Apple's code base/binaries makes me wonder how feasable replacing Apple's versions of libraries would be.

**Comment by tpoechtrager** [comment]
@BioMike: Yes, should be absolutely doable, but you would have to link against static libraries then, otherwise you may end up with ABI incompatibilities (or ship the dylibs with your application).

Btw, there is no need to cross-compile boost, you could use `osxcross-macports` instead:

 `osxcross-macports install boost` :smile:.

**Comment by geoff-codes** [comment]
FYI —

It is _theoretically possible_ to create to re-create "SDKs" via the "Current OS" SDKs (header packages, distributed in Software Updates), with other Software Updates + `strip -s` (is that the flag? the one that creates stub DMGs), without agreeing to the the Xcode EULA. I've done so a few times manually. If there's enough people who would wan't this, I'd consider writing a script to do this automatically. Thoughts.

**Comment by zarvox** [comment]
@geoff-codes I'm interested in knowing how you accomplished this.  Are those packages you repack into an SDK legally redistributable, and under what license?

Personally, I'd love to be able to redistribute packages, or at least have something to point automated build scripts toward that doesn't require hosting potentially infringing artifacts.  For instance, I think it'd be great to use osxcross and wclang to make buildbots that can produce binaries targeting all three major platforms.

**Comment by BioMike** [comment]
I don't think that is legal (or the moral right thing) to do. I assume the "Current OS" also has an EULA, which you agreed to.

**Comment by geoff-codes** [comment]
@zarvox
> Personally, I'd love to be able to redistribute packages, or at least have something to point automated build scripts toward that doesn't require hosting potentially infringing artifacts.

Probably not legally redistributable no, but not hosting potentially infringing artifacts, definitely; all the relevant code, etc. is pulled from publicly available software update servers over unsecured HTTP.

>  For instance, I think it'd be great to use osxcross and wclang to make buildbots that can produce binaries targeting all three major platforms.
This is exactly the use case I've done this with.

I'll post more details in the next couple of days with regards the process.

@BioMike I'd prefer not to get into the nitty-gritty of the legal argument here... what I will say is that while one arguably is bound to this EULA (±the standard Apple consumer EULA), it is significantly less stringent than the language in the [Xcode/ADC EULA](https://www.apple.com/legal/sla/docs/xcode.pdf) (see sections 2.2 and 2.5), which probably "applies" to the presently used methods of extracting SDKs.

**Comment by tpoechtrager** [comment]
I have tried a similar approach a while ago.

I have created C stub source files, which can be compiled to .dylibs, which then can be used to re-create the SDK in combination with the header files from [opensource.apple.com](https://opensource.apple.com/).

For example: http://git.io/vLpoP represents libc++.

But I threw it away rather quickly because the SDK doesn't make a lot of sense without all the proprietary stuff (frameworks, ...).

**Comment by barrystyle** [comment]
Wouldn't worry _too much_ regarding the legality of distributing Apple's SDK.

Mac OS X well and truly violates the BSD4.4 license agreement of which its entire codebase was developed from. You cannot take something that is open-source, with very specific conditions for redistribution and relicense it as proprietary software (with open source 'portions', give me a break).

Corporate thieves.
